sdk/core/System.ClientModel/src/Convenience/ClientResultOfT.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.ClientModel.Primitives;
+using System.ComponentModel;
+
+namespace System.ClientModel;
+
+public class ClientResult<T> : OptionalClientResult<T>
+{
+    protected internal ClientResult(T value, PipelineResponse response)
+        : base(value, response)
+    {
+        // Null values must use OptionalClientResult<T>
+        ClientUtilities.AssertNotNull(value, nameof(value));
```

Comment:
Should we throw the same descriptive exception here as we do [here](https://github.com/Azure/azure-sdk-for-net/pull/41016/files#diff-6ec35c4588f988e965f0d1e4de19275e6692aef7d588ce7d45d136c2c9569082R36-R37)?

Reply:
Happy to!